src/game.cpp

```diff
@@ -4143,10 +4143,9 @@ void game::debug()
                         p.weapon = p.ret_null;
                         break;
                     case D_ITEM_WORN: {
-                        auto filter = [this]( const item & it ) {
-                            return it.is_armor();
-                        };
-                        int item_pos = inv_for_filter( "Make target equip:", filter );
+                        int item_pos = inv_for_filter( "Make target equip:", []( const item_location &location ) {
+                            return location.get_item()->is_armor();
```

Comment:
While the explicit `get_item` is more informative, `item_location` has an `operator->` that may make it shorter.
There is no ambiguity regarding the types since lambdas require explicit types in arguments (ie. no `auto`s).